Mathlib/Tactic/SimplexCategory.lean

```diff
@@ -0,0 +1,85 @@
+/-
+Copyright (c) 2025 Nick Ward. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nick Ward
+-/
+import Mathlib.AlgebraicTopology.SimplexCategory.Defs
+import Mathlib.Util.Superscript
+
+/-! # Truncated simplex category notation
+
+We define the notation `⦋m⦌ₙ` for the `m`-dimensional simplex in the
+`n`-truncated simplex category. The truncation proof `p : m ≤ n` can also be
+provided using the syntax `⦋m, p⦌ₙ`. This notation is available with
+`open SimplexCategory.Truncated`.
+-/
+
+namespace SimplexCategory.Truncated
+
+open CategoryTheory
+open Mathlib.Tactic (subscriptTerm)
+
+/-- A quick attempt to prove that `⦋m⦌` is `n`-truncated (`⦋m⦌.len ≤ n`). -/
+scoped macro "trunc" : tactic =>
+  `(tactic| dsimp only [SimplexCategory.len_mk] <;> omega)
+
+/-- For `m ≤ n`, `⦋m⦌ₙ` is the `m`-dimensional simplex in `Truncated n`. The
+proof `p : m ≤ n` can also be provided using the syntax `⦋m, p⦌ₙ`. -/
+scoped syntax:max (name := mkNotation)
+  "⦋" term ("," term)? "⦌" noWs subscriptTerm : term
+scoped macro_rules
+  | `(⦋$m:term⦌$n:subscript) =>
+    `((⟨SimplexCategory.mk $m, by first | trunc |
+      fail "Failed to prove truncation property. Try writing `⦋m, by ...⦌ₙ`."⟩ :
+      SimplexCategory.Truncated $n))
+  | `(⦋$m:term, $p:term⦌$n:subscript) =>
+    `((⟨SimplexCategory.mk $m, $p⟩ : SimplexCategory.Truncated $n))
+
+section Delab
+open Mathlib.Tactic.Superscript (Mapping)
+open Lean PrettyPrinter.Delaborator SubExpr
+
+/-- Checks that the provided expression can be subscripted. -/
+private partial def subscriptable (e : Expr) : DelabM Unit := do
+  -- Any number or free variable with a subscriptable name is subscriptable.
+  if (← name e).any isSubscriptable || (← delab) matches `($_:num) then return
+  -- Addition and subtraction are subscriptable if their operands are.
+  guard <| e.isAppOfArity ``HAdd.hAdd 6 || e.isAppOfArity ``HSub.hSub 6
+  let #[_, _, _, _, x, y] := e.getAppArgs | failure
+  let _ ← withNaryArg 4 <| subscriptable x
+  let _ ← withAppArg <| subscriptable y
+where
+  -- Return the user-facing name of any constant or free variable.
+  name : Expr → MetaM (Option Name)
+    | Expr.const name _ => pure name
+    | Expr.fvar name => name.getUserName
+    | _ => pure none
+  -- Return `true` if every character in `s` can be subscripted.
+  isSubscriptable (s : Name) : Bool :=
+    s.toString.toList.all Mapping.subscript.toSpecial.contains
+
+/-- Checks that the provided expression can be subscripted before delaborating. -/
+def Meta.subscript (e : Expr) : Delab := subscriptable e >>= fun () ↦ delab
```

Comment:
This belongs in the same file as the subscript notation (or at least, next to it)

Reply:
Currently, this `SimplexCategory.Truncated.Meta.subscript` function is used in three different notations: `⦋m⦌ₙ` (defined in this file) as well as `X _⦋m⦌ₙ` and `X ^⦋m⦌ₙ` (defined in `SimplicialObject/Basic.lean`).

Would you suggest moving it to somewhere more generic, or is it fine to be defined in the same place as the first of these three notations? I can also add a reference to this function from the two delaborators in `SimplicialObject/Basic.lean`.

Reply:
I mean it should live next to the thing that implements the low level parser for subscripts (`subscriptTerm`?), much further upstream.

Reply:
Oh, I misunderstood. My mistake. 

My only concern is that `subscriptable` may not be general enough to live in `Mathlib.Util`. In particular, it doesn't yet handle function application, which may be valid in a subscript. I will move it and add an explanatory comment, though.

Would it be worth adding an analogous `superscriptable` function as well (perhaps in a separate PR)?

Reply:
I think all these new delaborator tools like `subscriptable` belong in a separate PR to this one, and at that point `superscrptable` should come along for the ride.

> In particular, it doesn't yet handle function application, which may be valid in a subscript.

I think it's fine as long as the function is overly conservative, and the docstring mentions where.

Reply:
There are some rough edges still, but #23018 provides a slightly more robust implementation of `subscriptable` and `superscriptable`.